xlsxwriter.class.php

```diff
@@ -257,10 +257,11 @@ public function writeSheetRow($sheet_name, array $row, $row_options=null)
 		
 		if (!empty($row_options))
 		{
-			$ht = isset($row_options['height']) ? floatval($row_options['height']) : 12.1;
-			$customHt = isset($row_options['height']) ? true : false;
-			$hidden = isset($row_options['hidden']) ? (bool)($row_options['hidden']) : false;
-			$collapsed = isset($row_options['collapsed']) ? (bool)($row_options['collapsed']) : false;
+			$row_settings = is_array($row_options[0]) ? $row_options[0] : $row_options;
+			$ht = isset($row_settings['height']) ? floatval($row_settings['height']) : 12.1;
```

Comment:
The default value can be a protected variable or have a setter and getter also in the constructor 
Check this fork https://github.com/elminson/PHP_XLSXWriter